File: src/main/java/com/example/RESTAPIarticle/service/MagazineServiceImpl.java
package com.example.RESTAPIarticle.service;

import com.example.RESTAPIarticle.dao.MagazineDAO;
import com.example.RESTAPIarticle.entity.Magazine;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class MagazineServiceImpl implements MagazineService{
    private final MagazineDAO magazineDAO;

    @Override
    public Magazine findById(int theId) {
        return magazineDAO.findById(theId).orElseThrow();
    }

    @Override
    public void save(Magazine theMagazine) {
        theMagazine.setId(0);
        magazineDAO.save(theMagazine);
    }

    @Override
    public Magazine findByName(String theName) {
        return magazineDAO.findByName(theName);
    }
}

File: src/main/java/com/example/RESTAPIarticle/service/MagazineServiceImpl.kt
package com.example.RESTAPIarticle.service

import com.example.RESTAPIarticle.entity.Magazine
import com.example.RESTAPIarticle.repository.MagazineRepository
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class MagazineServiceImpl(private val magazineRepository: MagazineRepository) : MagazineService {

    override fun findById(theId: Int): Magazine? {
        return magazineRepository.findById(theId).orElse(null)
    }

    override fun save(theMagazine: Magazine) {
        theMagazine.id = 0
        magazineRepository.save(theMagazine)
    }

    override fun findByName(theName: String): Magazine? {
        return magazineRepository.findByName(theName)
    }
}